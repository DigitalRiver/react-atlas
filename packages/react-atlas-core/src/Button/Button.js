import React from "react";
import PropTypes from "prop-types";
import {IconCore} from './../Icon';
import cx from "classnames";

/**
 * A Generic button component.
 */
class Button extends React.PureComponent {
  constructor(props) {
    super(props);
  }

  render() {
    const {
      onClick,
      className,
      children,
      disabled,
      large,
      small,
      primary,
      secondary,
      warning,
      error,
      link,
      outline,
      icon,
      type,
      href,
      style
    } = this.props;

    let mainStyle = "button";
    if (primary) {
      mainStyle = "primary";
    } else if (secondary) {
      mainStyle = "secondary";
    } else if (warning) {
      mainStyle = "warning";
    } else if (error) {
      mainStyle = "error";
    } else if (link) {
      mainStyle = "link";
    } else {
      mainStyle = "default_btn";
    }

    if (outline) {
      mainStyle += "_outline";
    }

    /**
     * Use cx to construct a CSS class object.
     * cx will automatically remove falsy values
     * for us. This way only passed props will be
     * transformed into CSS classes.
     */
    const classes = cx(
      {
        "disabled": disabled,
        "large": large,
        "small": small,
        "button": true
      },
      mainStyle
    );

    let text = children;
    let iconClass = false;

    /* Just use the icon instead of text when an icon
     * is passed and no children(text) is set. */
    if (icon && children === "Default Button") {
      text = null;
    } else if (icon) {
      text = children;
      iconClass = "ra_Button__icon-left";
    }

    let btn = <button
          onClick={onClick}
          className={cx(className)}
          styleName={classes}
          style={style}
          type={type}
          href={href}
        >
          {icon ? <IconCore icon={icon} /> : null}
          {text}
        </button>
    ;

    let renderButton = btn;
    if (href) {
      renderButton = <a href={href}>{btn}</a>;
    }

    return renderButton;
  }
}

Button.propTypes = {
  /** HTML5 button type, eg submit, button, etc */
  "type": PropTypes.string,
  /** The URL to link to. */
  "href": PropTypes.string,
  /**
   * Define a mini button.
   *
   */
  "children": PropTypes.node,
  /**
   * Click event handler.
   */
  "onClick": PropTypes.func,
  /** An Object, array, or string of CSS classes to apply to button.*/
  "className": PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object,
    PropTypes.array
  ]),
  /**
   *Determines if the button is disabled.
   * @examples <Button disabled/>
   */
  "disabled": PropTypes.bool,
  /**
   * use outline styled button
   * @examples <Button outline>
   */
  "outline": PropTypes.bool,
  /**
   * use primary style button
   * @examples <Button primary>
   */
  "primary": PropTypes.bool,
  /**
   * use secondary style button
   * @examples <Button secondary>
   */
  "secondary": PropTypes.bool,
  /**
   * use warning style button
   * @examples <Button warning>
   */
  "warning": PropTypes.bool,
  /**
   * use error style button
   * @examples <Button error>
   */
  "error": PropTypes.bool,
  /**
   * use link style button
   * @examples <Button link>
   */
  "link": PropTypes.bool,
  /**
   * use large style button
   * @examples <Button large>
   */
  "large": PropTypes.bool,
  /**
   * use small style button
   * @examples <Button small>
   */
  "small": PropTypes.bool,

  /**
   * The class name of the icon you want to set.
   */
  "icon": PropTypes.string,

  /**
   * Pass inline styles here.
   */
  "style": PropTypes.object
};

Button.defaultProps = {
  "children": "Default Button",
  "outline": false,
  "primary": false,
  "secondary": false,
  "warning": false,
  "error": false,
  "link": false,
  "large": false,
  "small": false,
  "disabled": false,
  "type": "button"
};

export default Button;
