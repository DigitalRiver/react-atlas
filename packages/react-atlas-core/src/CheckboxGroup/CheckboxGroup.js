import React, { cloneElement } from "react";
import PropTypes from "prop-types";
import cx from "classnames";

class CheckboxGroup extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      "totalChecked": this.initialChecked(),
      "groupError": false
    };
  }

  // Loop through child components to determine how many <Checkbox /> components are checked by default
  initialChecked() {
    let newChecked = 0;
    React.Children.map(this.props.children, child => {
      if (child.props.checked) {
        newChecked++;
      }
    });
    return newChecked;
  }

  // When child <Checkbox /> component changes state, update "totalChecked" value and verify against min or max props value
  handleChange = (value, event, isValid, checked) => {
    event.persist();
    const newChecked = checked
      ? this.state.totalChecked + 1
      : this.state.totalChecked - 1;
    if (checked) {
      this.setState({ "totalChecked": newChecked });
    } else {
      this.setState({ "totalChecked": newChecked });
    }
    if (
      this.props.max && newChecked > this.props.max ||
      this.props.min && newChecked < this.props.min
    ) {
      this.setState({ "groupError": true });
    } else {
      this.setState({ "groupError": false });
    }

    if (typeof this.props.onChange !== "undefined") {
      this.props.onChange(value, event, isValid, checked);
    }
  };

  // If min or max props are set and the correct number of checkboxes aren't checked, display the appropriate error message
  maxMinMessage = () => {
    let fullMessage;
    if (this.props.max && this.state.totalChecked > this.props.max) {
      if (this.props.limitMessage) {
        fullMessage = this.props.limitMessage.replace("{0}", this.props.max);
        return fullMessage;
      } else {
        return `Please select no more than ${this.props
          .max} of the options below`;
      }
    } else if (this.props.min && this.state.totalChecked < this.props.min) {
      if (this.props.limitMessage) {
        fullMessage = this.props.limitMessage.replace("{0}", this.props.min);
        return fullMessage;
      } else {
        return (
          "Please select at least " + this.props.min + " of the options below"
        );
      }
    } else {
      return false;
    }
  };

  render() {
    const {
      className,
      children,
      name,
      inline,
      inlineChildren,
      title,
      style
    } = this.props;
    const checkboxGroupStyles = cx({
      "checkboxGroup": !inline,
      "inline": inline
    });
    return (
      <div
        style={style}
        className={cx(className)}
        styleName={checkboxGroupStyles}
      >
        {title &&
          <div styleName={"header"}>
            <span styleName={"headerFont"}>{title}</span>
            {this.state.groupError &&
              <span styleName={"error_message"}>{this.maxMinMessage()}</span>
            }
          </div>
        }
        {React.Children.map(children, child => {
          child = cloneElement(child, {
            "inline": inlineChildren,
            "name": name,
            "onChange": this.handleChange,
            "groupError": this.state.groupError
          });
          return child;
        })}
      </div>
    );
  }
}

CheckboxGroup.propTypes = {
  /**
   * An optional callback that is fired when any checkbox in the
   * CheckboxGroup changes value. The callback has four arguments.
   * function(value, event, isValid, checked).
   */
  "onChange": PropTypes.func,
  /**
   * Anything that can be in a checkbox group. Typically only includes Checkbox components and a header.
   * @examples '<CheckboxGroup><Checkbox/><Checkbox/></CheckboxGroup>'
   */
  "children": PropTypes.node.isRequired,
  /** An Object, array, or string of CSS classes to apply to CheckboxGroup.*/
  "className": PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object,
    PropTypes.array
  ]),
  /**
   * Form name for the element, this will set all Checkbox children the same form name.
   * @examples '<CheckboxGroup name="test"></CheckboxGroup>'
   */
  "name": PropTypes.string,
  /**
   * Displayed header text for the CheckboxGroup.
   * @examples '<CheckboxGroup title="Options"></CheckboxGroup>'
   */
  "title": PropTypes.string,
  /**
   * Defines if the checkbox group should display as an inline element.
   * @examples '<CheckboxGroup inline></CheckboxGroup>'
   */
  "inline": PropTypes.bool,
  /**
   * Defines if the checkbox group should display child Checkbox components inline.
   * @examples '<CheckboxGroup inlineChildren></CheckboxGroup>'
   */
  "inlineChildren": PropTypes.bool,
  /**
   * Defines a minimum number of checkboxes the user must select in order to proceed.
   * @examples '<CheckboxGroup min="3"></CheckboxGroup>'
   */
  "min": PropTypes.number,
  /**
   * Defines a maximum number of checkboxes the user is allowed to select before proceeding.
   * @examples '<CheckboxGroup max="3"></CheckboxGroup>'
   */
  "max": PropTypes.number,
  /**
   * Defines a custom message to show if either min or max properties are not met. {0} can be used in the message to be replaced by the appropriate min or max property.
   * @examples '<CheckboxGroup min="3" limitMessage="Please select at least {0} of the checkboxes below."></CheckboxGroup>'
   */
  "limitMessage": PropTypes.string,

  /**
   * Pass inline styles here.
   */
  "style": PropTypes.object
};

CheckboxGroup.defaultProps = {
  "inline": false
};

export default CheckboxGroup;
