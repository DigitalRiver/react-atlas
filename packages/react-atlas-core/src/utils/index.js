import events from "./events";
import prefixer from "./prefixer";
// import time from './time';
import utils from "./utils";
import classNames from "./classNames";

export { events, prefixer, utils, classNames };
