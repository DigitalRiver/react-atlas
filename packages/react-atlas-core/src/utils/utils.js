export default {
  angleFromPositions(cx, cy, ex, ey) {
    const theta = Math.atan2(ey - cy, ex - cx) + Math.PI / 2;
    return theta * 180 / Math.PI;
  },
  angle360FromPositions(cx, cy, ex, ey) {
    const angle = this.angleFromPositions(cx, cy, ex, ey);
    return angle < 0 ? 360 + angle : angle;
  },
  range(start = 0, stop = null, step = 1) {
    let [_start, _stop] = [0, start];
    if (stop !== null) {
      [_start, _stop] = [start, stop];
    }
    const length = Math.max(Math.ceil((_stop - _start) / step), 0);
    const range = Array(length);

    for (let idx = 0; idx < length; idx++, _start += step) {
      range[idx] = _start;
    }

    return range;
  },
  round(number, decimals) {
    if (!isNaN(parseFloat(number)) && isFinite(number)) {
      const decimalPower = Math.pow(10, decimals);
      return Math.round(parseFloat(number) * decimalPower) / decimalPower;
    }
    return NaN;
  },
  getViewport() {
    return {
      "height": window.innerHeight || document.documentElement.offsetHeight,
      "width": window.innerWidth || document.documentElement.offsetWidth
    };
  },
  cloneObject(object) {
    return JSON.parse(JSON.stringify(object));
  },
  inputTypeForPrototype(prototype) {
    if (prototype === Date) {
      return "date";
    }
    if (prototype === Number) {
      return "number";
    }
    if (prototype === Boolean) {
      return "checkbox";
    }
    return "text";
  },
  prepareValueForInput(value, type) {
    if (type === "date") {
      return new Date(value).toISOString().slice(0, 10);
    }
    if (type === "checkbox") {
      return value ? "on" : null;
    }
    return value;
  },
  getSelection(elem) {
    let start, end, rangeElem, clone;

    if (elem.selectionStart !== undefined) {
      start = elem.selectionStart;
      end = elem.selectionEnd;
    }
    else {
      try {
        elem.focus();
        rangeElem = elem.createTextRange();
        clone = rangeElem.duplicate();

        rangeElem.moveToBookmark(document.selection.createRange().getBookmark());
        clone.setEndPoint('EndToStart', rangeElem);

        start = clone.text.length;
        end = start + rangeElem.text.length;
      }
      catch (e) { /* not focused or not visible */ }
    }

    return { start, end };
  },
  setSelection(elem, selection) {
    let rangeElem;

    try {
      if (elem.selectionStart !== undefined) {
        elem.focus();
        elem.setSelectionRange(selection.start, selection.end);
      }
      else {
        elem.focus();
        rangeElem = elem.createTextRange();
        rangeElem.collapse(true);
        rangeElem.moveStart('character', selection.start);
        rangeElem.moveEnd('character', selection.end - selection.start);
        rangeElem.select();
      }
    }
    catch (e) { /* not focused or not visible */ }
  }
};
