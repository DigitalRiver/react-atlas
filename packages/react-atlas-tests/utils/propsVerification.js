export function verifyPropsDefaultValue(component, expectedProps) {
  //Compares each property value of the component with the expected value

  var result = true;
  var partialResult;

  for (var [compPropName, compPropValue] of expectedProps.entries()) {
    partialResult = component.props()[compPropName] == compPropValue;

    if (!partialResult) {
      console.log(
        "[" +
          compPropName +
          "] -> Got: " +
          component.props()[compPropName] +
          " - expected: " +
          compPropValue
      );
      result = false;
    }
  }
  return result;
}
