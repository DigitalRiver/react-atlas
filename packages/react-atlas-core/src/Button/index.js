import { default as ButtonCore } from "./Button";
export { ButtonCore };
