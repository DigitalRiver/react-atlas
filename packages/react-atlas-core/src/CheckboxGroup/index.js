import { default as CheckboxGroupCore } from "./CheckboxGroup";
export { CheckboxGroupCore };