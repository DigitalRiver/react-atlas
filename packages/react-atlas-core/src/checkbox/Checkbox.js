import React, { Component, PropTypes } from 'react';
import themeable from 'react-themeable';
import { classNames } from '../utils/utils';


/**
 * Simple component for a basic checkbox
 */
const Checkbox = ({title, label, disabled, inline, ...other}) => {
      //const cx = ClassNames.bind(other.theme);
      const theme = themeable(other.theme);
      const componentClasses = classNames({
          "block": !inline,
          inline,
          disabled,
          [className]: className
      });

      title = title ? title : label;

      return (
          <label {...theme(1, ...componentClasses)} title={title}>
              <input {...other} type="checkbox" disabled={disabled} {...theme(2, 'input')} />
              {label && <span {...theme(3, 'label')}>{label}</span>}
          </label>
      );
};

Checkbox.propTypes = {
  /**
   * Text for checkbox label
   * @examples 'Some Label'
   */
  label: PropTypes.string,
	/**
   * A css class name that will be appended to the wrapping <label> element around the <input> and <span> elements.
   */
  className: PropTypes.string,
	/**
   * If included, checkbox is disabled
   * @examples <Checkbox disabled />, <Checkbox disabled={true} />
   */
  disabled: PropTypes.bool,
	/**
   * If included, renders the checkbox and it's label inline, so it can be side-by-side to other content.
   */
  inline: PropTypes.bool,
  /**
   * Text for checkbox label title. (i.e. "alt-text" for checkboxes, useful for accessibility). If not provided, will be label text.
   * @examples 'Some Title'
   */
  title: PropTypes.string,
	/**
   * Defines if checkbox should be checked on load.
   */
  defaultChecked: PropTypes.bool
};

Checkbox.defaultProps = {
  className: '',
  disabled: false,
  inline: false
};

Checkbox.styleguide = {
  category: "Form Components",
  index: "3.2",
  example: `
<section style={{textAlign:"left"}}>
  <h4 style={{marginBottom: "10px"}}>Checkboxes</h4>
  <Checkbox
    defaultChecked={true}
    label="Checked"
  />
  <Checkbox
    defaultChecked={false}
    label="Not checked"

  />
  <Checkbox
    defaultChecked={true}
    label="Disabled checkbox"
    disabled
  />
  <Checkbox
    defaultChecked={false}
    label="Inline checkbox 1"
    inline
  />
  <Checkbox
    defaultChecked={false}
    label="Inline checkbox 2"
    inline
  />
</section>
`
};

export default Checkbox;
