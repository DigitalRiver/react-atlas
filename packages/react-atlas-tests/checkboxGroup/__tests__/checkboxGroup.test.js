import React from "react";
import { mount, shallow } from "enzyme";
import { CheckboxGroupCore } from "../../../react-atlas-core/src/CheckboxGroup/index";
import { CheckboxCore } from "../../../react-atlas-core/src/Checkbox/index";
import { verifyPropsDefaultValue } from "../../utils/propsVerification";

import { default as Button } from "../../../react-atlas-core/src/button/Button";

describe("Test CheckboxGroup component - the bascis", () => {
  it("Checkbox group - 4 checkboxes, 1 checked", function() {
    const component = mount(
      <CheckboxGroupCore title={"CheckboxGroup test"}>
        <CheckboxCore name={"1"} />
        <CheckboxCore name={"2"} />
        <CheckboxCore name={"3"} />
        <CheckboxCore name={"4"} />
      </CheckboxGroupCore>
    );

    component.childAt(1).simulate("click");

    expect(component.state().totalChecked).toBe(1);
  });
});

describe("Test CheckboxGroup component - the bascis", () => {
	it("Checkbox group - 4 checkboxes, 1 checked", function(){
		const component = mount(<CheckboxGroupCore title={'CheckboxGroup test'}>
									<CheckboxCore name={'1'}/>
		                            <CheckboxCore name={'2'}/>
									<CheckboxCore name={'3'}/>
									<CheckboxCore name={'4'}/>
								</CheckboxGroupCore>);

		component.childAt(1).simulate('click');

		expect(component.state().totalChecked).toBe(1);
	});

	it("Checkbox group - 4 checkboxes, 2 checked 1 from creation", function() {
		const component = mount(<CheckboxGroupCore title={'CheckboxGroup test'}>
									<CheckboxCore name={'1'} checked={true}/>
		                            <CheckboxCore name={'2'}/>
									<CheckboxCore name={'3'}/>
									<CheckboxCore name={'4'}/>
								</CheckboxGroupCore>);

		component.childAt(2).simulate('click');

		expect(component.state().totalChecked).toBe(2);
	});

	it("Checkbox group - 4 checkboxes, 0 checked with 1 checked from creation", function() {
		const component = mount(<CheckboxGroupCore title={'CheckboxGroup test'}>
									<CheckboxCore name={'1'}/>
		                            <CheckboxCore name={'2'}/>
									<CheckboxCore name={'3'}/>
									<CheckboxCore name={'4'}/>
								</CheckboxGroupCore>);

		component.childAt(0).simulate('click');

		expect(component.state().totalChecked).toBe(0);
	});

	it("Checkbox group - 4 checkboxes, 3 checked", function() {
		const component = mount(<CheckboxGroupCore title={'CheckboxGroup test'}>
									<CheckboxCore name={'1'}/>
		                            <CheckboxCore name={'2'}/>
									<CheckboxCore name={'3'}/>
									<CheckboxCore name={'4'}/>
								</CheckboxGroupCore>);

		component.childAt(1).simulate('click');
		component.childAt(2).simulate('click');
		component.childAt(3).simulate('click');

		expect(component.state().totalChecked).toBe(3);
	});

	it("Checkbox group - 4 checkboxes, 2 checked", function() {
		const component = mount(<CheckboxGroupCore title={'CheckboxGroup test'}>
									<CheckboxCore name={'1'}/>
		                            <CheckboxCore name={'2'}/>
									<CheckboxCore name={'3'}/>
									<CheckboxCore name={'4'}/>
								</CheckboxGroupCore>);

		component.childAt(1).simulate('click');
		component.childAt(2).simulate('click');
		component.childAt(3).simulate('click');

		expect(component.state().totalChecked).toBe(3);

		component.childAt(1).simulate('click');

		expect(component.state().totalChecked).toBe(2);
	});
});

function _verifyMinMax(mi, ma, numberOfClicks, message, expectedResult)	{
	const component = mount(<CheckboxGroupCore title={'CheckboxGroup test'} min={mi} max={ma} >
								<CheckboxCore name={'1'}/>
								<CheckboxCore name={'2'}/>
								<CheckboxCore name={'3'}/>
								<CheckboxCore name={'4'}/>
								<CheckboxCore name={'5'}/>
							</CheckboxGroupCore>);
	if(message != ''){
		component.limitMessage = message;
	}
	for(let i=1; i<=numberOfClicks; i++){
		component.childAt(i).simulate('click');
	}

	expect(component.state().groupError).toBe(expectedResult);
}

describe("Test CheckboxGroup component - Min/Max tests", () => {
	it("Checkbox group - 5 checkboxes, max=4, min=2, 1 checked", function() { _verifyMinMax(2, 4, 1, 'Limit exceed!!', true); });
	it("Checkbox group - 5 checkboxes, max=4, min=2, 1 checked", function() { _verifyMinMax(2, 4, 2, 'Limit exceed!!', false); });
	it("Checkbox group - 5 checkboxes, max=4, min=2, 3 checked", function() { _verifyMinMax(2, 4, 3, 'Limit exceed!!', false); });
	it("Checkbox group - 5 checkboxes, max=4, min=2, 1 checked", function() { _verifyMinMax(2, 4, 4, 'Limit exceed!!', false); });
	it("Checkbox group - 5 checkboxes, max=4, min=2, 5 checked", function() { _verifyMinMax(2, 4, 5, 'Limit exceed!!', true); });
	it("Checkbox group - 5 checkboxes, max=4, min=2, 1 checked no Limmit message", function() { _verifyMinMax(2, 4, 1, '', true); });
	it("Checkbox group - 5 checkboxes, max=4, min=2, 5 checked no Limmit message", function() { _verifyMinMax(2, 4, 5, '', true); });

	it("Checkbox group - 5 checkboxes, max=4, min=2, 3 checked", function() {
		const component = mount(<CheckboxGroupCore title={'CheckboxGroup test'} max={4} min={2} limitMessage={'Limit exceed!!'} >
									<CheckboxCore name={'1'}/>
		                            <CheckboxCore name={'2'}/>
									<CheckboxCore name={'3'}/>
									<CheckboxCore name={'4'}/>
									<CheckboxCore name={'5'}/>
								</CheckboxGroupCore>);

		for(let i=1; i<=5; i++){
			component.childAt(i).simulate('click');
		}

		expect(component.state().groupError).toBe(true);

		component.childAt(2).simulate('click');
		component.childAt(0).simulate('click');

		expect(component.state().groupError).toBe(false);
	});
});
