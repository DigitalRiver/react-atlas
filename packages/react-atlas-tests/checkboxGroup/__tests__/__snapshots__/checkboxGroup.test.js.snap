// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CheckboxGroup component - the basics Test renders correctly 1`] = `
<div
  className=""
  style={undefined}
  styleName="checkboxGroup"
>
  <div
    styleName="header"
  >
    <span
      styleName="headerFont"
    >
      CheckboxGroup test
    </span>
  </div>
  <div
    className=""
    onClick={[Function]}
    style={undefined}
    styleName="checkbox_padding"
  >
    <div
      styleName="inline_block relative padding"
    >
      <div
        styleName="float_left"
      >
        <input
          checked={false}
          className="ra_Input__checkbox ra_styles__marg-b-1 ra_Input__max ra_Input__opacity"
          id={undefined}
          label={undefined}
          name={undefined}
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onPaste={[Function]}
          required={false}
          style={undefined}
          type="checkbox"
        />
        <div
          styleName="not_checked"
        />
      </div>
    </div>
  </div>
  <div
    className=""
    onClick={[Function]}
    style={undefined}
    styleName="checkbox_padding"
  >
    <div
      styleName="inline_block relative padding"
    >
      <div
        styleName="float_left"
      >
        <input
          checked={false}
          className="ra_Input__checkbox ra_styles__marg-b-1 ra_Input__max ra_Input__opacity"
          id={undefined}
          label={undefined}
          name={undefined}
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onPaste={[Function]}
          required={false}
          style={undefined}
          type="checkbox"
        />
        <div
          styleName="not_checked"
        />
      </div>
    </div>
  </div>
  <div
    className=""
    onClick={[Function]}
    style={undefined}
    styleName="checkbox_padding"
  >
    <div
      styleName="inline_block relative padding"
    >
      <div
        styleName="float_left"
      >
        <input
          checked={false}
          className="ra_Input__checkbox ra_styles__marg-b-1 ra_Input__max ra_Input__opacity"
          id={undefined}
          label={undefined}
          name={undefined}
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onPaste={[Function]}
          required={false}
          style={undefined}
          type="checkbox"
        />
        <div
          styleName="not_checked"
        />
      </div>
    </div>
  </div>
  <div
    className=""
    onClick={[Function]}
    style={undefined}
    styleName="checkbox_padding"
  >
    <div
      styleName="inline_block relative padding"
    >
      <div
        styleName="float_left"
      >
        <input
          checked={false}
          className="ra_Input__checkbox ra_styles__marg-b-1 ra_Input__max ra_Input__opacity"
          id={undefined}
          label={undefined}
          name={undefined}
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onPaste={[Function]}
          required={false}
          style={undefined}
          type="checkbox"
        />
        <div
          styleName="not_checked"
        />
      </div>
    </div>
  </div>
</div>
`;
